src/main/haskell/kore/test/Test/Kore/Step/MockSymbols.hs

```diff
@@ -985,6 +1046,15 @@ attributesMapping =
     ,   ( sortInjectionSubSubToSubSymbol
         , Mock.sortInjectionAttributes
         )
+    ,   ( unitMapSymbol
+        , Mock.defaultAttributes { hook = Hook (Just "MAP.unit") }
```

Comment:
In the instance we get from `K`, it is `functional`.  In some sense, it is a hooked function, as it is evaluating to its internal representation.   And, given our current accepted convention that you cannot have both domain values and constructors on the same sort, I guess it cannot be a constructor.  One question is, then: how do we match on it?

Reply:
Ok, fair enough, it can't be a constructor. But, I guess, it should always be a functional symbol, not only in the K case (does it make any sense otherwise? you could never have an `x=Map.unit` substitution if it's not, at least, a function...), so then it should also be functional in our tests.

Reply:
In an earlier discussion about builtins, we said that the backend does not provide any axioms for builtins. Now it seems like we don't want that to be true.

In the specific example of `x = Map.unit()`, the builtin function `Map.unit()` will be evaluated to a builtin domain value, which is functional, so that the equality proceeds.

Reply:
Ok, I see, I guess that this is fine, then.